### PR TITLE
Implemented atomic representation for a single element type and valence electron only representation

### PR DESCRIPTION
**Pull request by YAY-C**
- **Fix element types ordering (moved to func)**
- **Fix output name for singlet open-shell**
- **implemented single atom type rep generation**
- **Implemented valence-onyl atom density rep**
- **Fixed argument passing**
- **Refix argument-passing bug in 5653500**

### CODE REVIEW

**Review by briling — CHANGES_REQUESTED** [review]
We need tests!

**Comment by YAY-C** [comment]
I added test units for the developped features within `test_spahm_a.py`.

**Comment by YAY-C** [comment]
Sorry forgot to push the test-data, tests should be fine now

**Review by briling — COMMENTED** [review]
1) I don't get these lines:
```
        if valence_only:
            a_dm1[start:stop,start:stop] = 0
```

2) if in `test_water_single_element()` remove `only_z=['O']` the test still passes

**Comment by YAY-C** [comment]
I explicitly check for the array `shape` within the test, that should fix the issue.

**Comment by briling** [comment]
> > 2. if in `test_water_single_element()` remove `only_z=['O']` the test still passes
> 
> I explicitly check for the array `shape` within the test, that should fix the issue.

cool thanks

**Comment by briling** [comment]
maybe remove the wrong valence only and have it in another pr? if we even need it

**Review by YAY-C — COMMENTED** [review]
> maybe remove the wrong valence only and have it in another pr? if we even need it

I moved all the changes for the valence stuff (now in #44),
and kept the single-element stuff in here.

I also added a few changes to get the `ecp`  parameter from the CLI `parser`.
_Should it be in a different PR?_

**Comment by YAY-C** [comment]
> I also added a few changes to get the ecp parameter from the CLI parser.

 I just realized I should do the same for spahm-b, let me know if you want both in  a new PR!
 
 And I was thinking maybe in the `main()` we should enforce and warn about the need for `ecp` argument when using minao with elements heavier than Y.
_What do you think?_

File: qstack/spahm/rho/atom.py
Comment (briling) [inline qstack/spahm/rho/atom.py:1]: maybe this
```
    for j,i in enumerate(only_i):
        q = mol.elements[i]
        v = rep[j]
```
can be replaced with this
```
    for q, v in zip(mol.elements[only_i], rep):
```
which is easier to read and more similar to the previous version

**Comment by briling** [comment]
> I just realized I should do the same for spahm-b, let me know if you want both in a new PR!
> 
> And I was thinking maybe in the `main()` we should enforce and warn about the need for `ecp` argument when using minao with elements heavier than Y. _What do you think?_

yes to both! but yeah probably these ecp stuff should be separate

**Review by briling — CHANGES_REQUESTED** [review]

File: qstack/spahm/rho/atom.py
Comment (YAY-C) [reply qstack/spahm/rho/atom.py:1]: I liked the one line. but it doesn't work cause `mol.elements` is a `list` and not a `numpy.array` so you can not subscript it with a list of integers.
_the intermediate option does not work neither then!_

I can create a copy of `mol.elements` in a `numpy.array` and go for the oneliner, but it feels a bit like an overhead.
_what do you think?_

**Review by YAY-C — COMMENTED** [review]
- feels like less trouble to leave it like this:
```
    for j,i in enumerate(only_i):
        q = mol.elements[i]
        v = rep[j]
```
- and I moved the ECP changes to a new issue/branch #48

File: qstack/spahm/rho/atom.py
Comment (briling) [reply qstack/spahm/rho/atom.py:1]: i'd put `zip(np.array(mol.elements)[only_i], rep)]` lol

**Comment by briling** [comment]
at this point there's so many commits and reverts that I'd squash and merge rather than merge

File: qstack/spahm/rho/atom.py
Comment (briling) [reply qstack/spahm/rho/atom.py:1]: I just don't like j,i

Comment (YAY-C) [reply qstack/spahm/rho/atom.py:1]: ahah sure! Done 👌

**Review by YAY-C — COMMENTED** [review]
sinlge-line code

**Comment by YAY-C** [comment]
> at this point there's so many commits and reverts that I'd squash and merge rather than merge

agreed, "17 commits" sounds too much for so few changes in the end!

I have never done it though, you wan to do it?

**Pull request merged**